### ISSUE TITLE
 Replacement of the original uws

### ISSUE THREAD

**Issue by Plutoject** [ClusterWS/ClusterWS-uWS#2]
Hi, can I use this library as a complete replacement for the original uws?
[https://www.npmjs.com/package/uws](url)

**Comment by goriunov** [comment]
Yes it should work but you may need to change some names as in readme for WebSocket Server and Client. Also this library does not have old way of implementing emitters:

```js
// this will not work
ws.onopen = function () {
}

// This will work
ws.on('open')
```

I will collect the difference and placing it to the docs shortly (hopefully by the end of this week).

This is not exactly drop in replacement as i have changed some things to make it look nicer and remove some things which are so much needed (with new js and conventions). But the main core is pretty much the same.

P.S.
I have modified only ping pong system to drop the connection if it does not get pong back from the client. And added app level ping pong which is really good for browsers clients (as they don't have on ping method) you will need to implement custom response for the ping on the client side.

This all info will go in to the docs soon. :)

**Comment by Plutoject** [comment]
it's very good, I'll do it. I already tried to put your library instead of ws and everything worked perfectly. But I found several problems related to the description of types
1) There are no types for config
```typescript
export class WebSocketServer extends EventEmitter {
    constructor(configs: any, callback: any);
```
2) I suppose there should be a number, not a string
```typescript
startAutoPing(interval: string, appLevel?: boolean): void;
```
3) For socket and request types, I used types from ws lib. Although I think in the future they will be exported with your library

4) Typescript compiler throws an error

node_modules/clusterws-uws/dist/index.d.ts(26,33): error TS2304: Cannot find name 'BinaryType'.

I do not exclude that the problem with my configuration
